### PR TITLE
Switch to libtirpc and support IPv6

### PR DESCRIPTION
**Pull request by skoudmar**
I have tested the pull request #17 and fixed and extended some of its functionality.

Works when connecting from both IPv4 and IPv6 clients.

Resolves:
#12, #13 

Helped by: @wentasah

### CODE REVIEW

**Comment by CendioOssman** [comment]
Thanks for testing!

Which of these are bug fixes for #17? I'd prefer to get that fully fixed and merged first, before looking at further improvements..

**Comment by skoudmar** [comment]
Commit f7acfc5 adds option `-4` which allowed me to find bugs related to starting unfsd on systems with IPv6 disabled (improvement). But the commit also fixes bug: Starting unfsd without specifying `-l <addr>` would crash on such systems.

Other commits are pure bug fixes.

**Review by CendioOssman — CHANGES_REQUESTED** [review]
I've incorporated the fixes in to my PR now. Thanks.

What remains here is the `-4` flag, and the portmap stuff, which I'm a bit unclear about.

File: daemon.c [hunk 1]
```diff
@@ -673,6 +702,12 @@ static void register_nfs_service(SVCXPRT * udptransp, SVCXPRT * tcptransp)
 		    "unable to register (NFS3_PROGRAM, NFS_V3, udp).");
 	    daemon_exit(0);
 	}
+
+	if (opt_portmapper) {
+	    /* Register NFS service for UDP6 */
+	    register_service(udptransp, NFS3_PROGRAM, NFS_V3, nfs3_program_3,
+		"udp6", "(NFS3_PROGRAM, NFS_V3, udp6)");
+	}
```
[the remark this answers is not in the document]
Comment (skoudmar) [reply daemon.c:745]: I don't see this done implicitly.

It adds mappings required if we want to connect using IPv6 address.

Example mounting command:
```
mount [::1]:/exported /mnt
```

Without the mapping the command fails with:
```
mount.nfs: requested NFS version or transport protocol is not supported
```

[the remark this answers is not in the document]
Comment (CendioOssman) [reply daemon.c:745]: Sorry for the long delay replying.

I redid the tests, and could no longer get it working. So I'm not sure what I did last time...

Your suggestion is entirely correct. However, it highlighted that `svc_register()` was a legacy interface, so I went a bit further in e02a907b8b4d431e68aebe6602aa3690cce525da and fe3210b67f18e231c18cda77637689e5979019d5.

**Comment by CendioOssman** [comment]
I've merged #17 now, and all your fixes should hopefully be included.

Do you think you could do a test as well to make sure something isn't missing?

**Comment by skoudmar** [comment]
Unfortunately, the functions svc_dg_create and svc_vc_create have different requirements on the state of the socket. More info and fix is in commit skoudmar/unfs3@a7bd4ba0c228cca1dbc78b8bf4699107a3542eaa.

**Comment by CendioOssman** [comment]
Crap. You're right. I missed that case. Thanks for the fix! It's on `master` now.

**Pull request closed without merging**